crates/storage/provider/src/traits/block.rs

```diff
@@ -6,6 +7,29 @@ use reth_storage_errors::provider::ProviderResult;
 use reth_trie::{updates::TrieUpdates, HashedPostStateSorted};
 use std::ops::RangeInclusive;
 
+/// An enum that represents the storage location for a piece of data.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum StorageLocation {
+    /// Write only to static files.
+    StaticFiles,
+    /// Write only to the database.
+    Database,
+    /// Write to both the database and static files.
+    Both,
```

Comment:
does this mean data is duplicated?

not obvious where we need this

Reply:
right now we are using `insert_block` which does this by default and on reorgs we use `take_block_transactions_range` which relies on all transactions being present in database https://github.com/paradigmxyz/reth/blob/b8980e5982e1d16adadee00230c090f39d8076be/crates/storage/provider/src/providers/database/provider.rs#L880

to get rid of this we'd need a separate fn for reorgs in new engine which I am planning to do in a follow-up